in_toto/gpg/common.py

```diff
@@ -334,11 +337,61 @@ def _assign_certified_key_info(bundle):
             "by '{}'.".format(signature["keyid"]))
         continue
 
-      # TODO: If the signature is valid, extract relevant information from
-      # its "info" field and assign to master key (e.g. expiration) here
-      # NOTE: Beware of conflicting information. There might be multiple User
-      # IDs per primary key and multiple signatures per User ID. See RFC4880
-      # 5.2.3.19. and last paragraph of 5.2.3.3. for more info about ambiguity.
+      # If the signature is valid, we try to extract subpackets relevant to
+      # the primary key, i.e. expiration time.
+      # NOTE: There might be multiple User IDs per primary key and multiple
+      # certificates per User ID. RFC4880 5.2.3.19. and last paragraph of
+      # 5.2.3.3. provides some suggestions about ambiguity, but delegates the
+      # responsibility to the implementer.
+
+      # Ambiguity resolution scheme:
+      # We take the key expiration time from the most recent certificate, i.e.
+      # the certificate with the highest signature creation time. Additionally,
+      # we prioritize certificates with primary user id flag set True. Note
+      # that, if the ultimately prioritized certificate does not have a key
+      # expiration time subpacket, we don't assign one, even if there were
+      # certificates of lower priority carrying that subpacket.
+      tmp_validity_period = \
```

Comment:
Is there a reason to rename `key_expiration` to `validity_period`? I think it may be confusing. What do you think?

Reply:
As per the [RFC 4880] (https://tools.ietf.org/html/rfc4880#section-5.2.3.6):
> **5.2.3.6.  Key Expiration Time**
> The validity period of the key.  This is the number of seconds after the key creation time that the key expires. [...]

So the spec calls it both. But IMO "expiration" or "expiration time" is more ambiguous. It could easily be both, the number of seconds between creation and expiration (which it is), or a set date on which it expires (which it isn't). I think `validity_period` is less ambiguous in that regard. 

That said, I'm not overly passionate about it. :)

Reply:
Hmm, I see what you mean. I just found myself jumping where `tmp_validity_period` was defined and where it was checked, as I would read on the dict the `KEY_EXPIRATION_SUBPACKET`.

I know the RFC is not the clearest piece of text ever written (just wait for v5 keys!) but yeah...